backend/src/apiserver/template/v2_template.go

```diff
@@ -306,3 +311,55 @@ func IsPlatformSpecWithKubernetesConfig(template []byte) bool {
 	_, ok := platformSpec.Platforms["kubernetes"]
 	return ok
 }
+
+func (t *V2Spec) validatePipelineJobInputs(job *pipelinespec.PipelineJob) error {
+	// The pipeline requires no input
```

Comment:
If the pipeline has no input, we should validate that runtime config also doesn't contain any parameter.

Reply:
I think as long as the users provided the parameters we need, it's ok if there's more. This makes our verification simpler and gives more leeway for the users. WDYT?

Reply:
I did borrow this logic from Vertex. They only verify that the needed parameters are provided, and do not care about extra parameters users give. But I think you have a good point that this may affect the speed of users' debugging.